ZLS build fails with error in `Thread.Pool` function pointer type mismatch

### Zig Version

0.14.0-dev.208+854e86c56

### Steps to Reproduce and Observed Behavior

Check out https://github.com/zigtools/zls/commit/fbd8b9a87a3cd5ab8389054815d9e3f81dfd430b and run `zig build`:

```
install
└─ install zls
   └─ zig build-exe zls Debug native 2 errors
/home/***/Software/Install/lib/zig/std/Thread/Pool.zig:224:50: error: expected type '*const fn (*Thread.Pool.Runnable, ?usize) void', found '*const fn (*Thread.Pool.Runnable) void'
        run_node: RunQueue.Node = .{ .data = .{ .runFn = runFn } },
                                                ~^~~~~~~~~~~~~
/home/***/Software/Install/lib/zig/std/Thread/Pool.zig:224:50: note: pointer type child 'fn (*Thread.Pool.Runnable) void' cannot cast into pointer type child 'fn (*Thread.Pool.Runnable, ?usize) void'
/home/***/Software/Install/lib/zig/std/Thread/Pool.zig:224:50: note: function with 1 parameters cannot cast into a function with 2 parameters
referenced by:
    spawn__anon_14882: /home/***/Software/Install/lib/zig/std/Thread/Pool.zig:221:21
    loop: src/Server.zig:1763:49
    4 reference(s) hidden; use '-freference-trace=6' to see all references
/home/***/Software/Install/lib/zig/std/Thread/Pool.zig:224:50: error: expected type '*const fn (*Thread.Pool.Runnable, ?usize) void', found '*const fn (*Thread.Pool.Runnable) void'
        run_node: RunQueue.Node = .{ .data = .{ .runFn = runFn } },
                                                ~^~~~~~~~~~~~~
/home/***/Software/Install/lib/zig/std/Thread/Pool.zig:224:50: note: pointer type child 'fn (*Thread.Pool.Runnable) void' cannot cast into pointer type child 'fn (*Thread.Pool.Runnable, ?usize) void'
/home/***/Software/Install/lib/zig/std/Thread/Pool.zig:224:50: note: function with 1 parameters cannot cast into a function with 2 parameters
error: the following command failed with 2 compilation errors:
/home/***/Software/Install/bin/zig build-exe -ODebug --dep exe_options --dep tracy --dep diffz --dep known-folders --dep zls -Mroot=/home/***/Projects/zls/src/main.zig -Mexe_options=/home/***/Projects/zls/.zig-cache/c/955b0f6de7d3d0cec574752b5067abd5/options.zig -ODebug --dep options -Mtracy=/home/***/Projects/zls/src/tracy.zig -Mdiffz=/home/***/.cache/zig/p/1220102cb2c669d82184fb1dc5380193d37d68b54e8d75b76b2d155b9af7d7e2e76d/DiffMatchPatch.zig -Mknown-folders=/home/***/.cache/zig/p/12209d2738a2e1dbd3781c2e5f01a2ea877dcfeea53efdfa1913247297d328e6b207/known-folders.zig --dep known-folders --dep diffz --dep tracy --dep build_options --dep version_data -Mzls=/home/***/Projects/zls/src/zls.zig -Moptions=/home/***/Projects/zls/.zig-cache/c/27d5d436158affe42215ba8eab749e7c/options.zig -Mbuild_options=/home/***/Projects/zls/.zig-cache/c/068f0d792067ad675740b4449faa87e7/options.zig -Mversion_data=/home/***/Projects/zls/.zig-cache/o/c8e09a9849209757d6307956a9d2366b/version_data_master_19913.zig --cache-dir /home/***/Projects/zls/.zig-cache --global-cache-dir /home/***/.cache/zig --name zls --listen=- 
Build Summary: 5/8 steps succeeded; 1 failed (disable with --summary none)
install transitive failure
└─ install zls transitive failure
   └─ zig build-exe zls Debug native 2 errors
error: the following build command failed with exit code 1:
/home/***/Projects/zls/.zig-cache/o/23a91e621a7c6e4c1b221792ead87157/build /home/***/Software/Install/bin/zig /home/***/Projects/zls /home/***/Projects/zls/.zig-cache /home/***/.cache/zig --seed 0x3f5c4ce1 -Za6cd173f145e6380

```

### Expected Behavior

ZLS builds successfully, or at least without an error in the standard library.

--- a/lib/std/Thread.zig
+++ b/lib/std/Thread.zig
@@ -1465,6 +1465,7 @@ test {
     _ = Semaphore;
     _ = Condition;
     _ = RwLock;
+    _ = Pool;
 }
 
 fn testIncrementNotify(value: *usize, event: *ResetEvent) void {

--- a/lib/std/Thread/Pool.zig
+++ b/lib/std/Thread/Pool.zig
@@ -254,6 +254,27 @@ pub fn spawn(pool: *Pool, comptime func: anytype, args: anytype) !void {
     pool.cond.signal();
 }
 
+test spawn {
+    const TestFn = struct {
+        fn checkRun(completed: *bool) void {
+            completed.* = true;
+        }
+    };
+
+    var completed: bool = false;
+
+    {
+        var pool: Pool = undefined;
+        try pool.init(.{
+            .allocator = std.testing.allocator,
+        });
+        defer pool.deinit();
+        try pool.spawn(TestFn.checkRun, .{&completed});
+    }
+
+    try std.testing.expectEqual(true, completed);
+}
+
 fn worker(pool: *Pool) void {
     pool.mutex.lock();
     defer pool.mutex.unlock();

--- a/lib/std/Thread/Pool.zig
+++ b/lib/std/Thread/Pool.zig
@@ -223,7 +223,7 @@ pub fn spawn(pool: *Pool, comptime func: anytype, args: anytype) !void {
         pool: *Pool,
         run_node: RunQueue.Node = .{ .data = .{ .runFn = runFn } },
 
-        fn runFn(runnable: *Runnable) void {
+        fn runFn(runnable: *Runnable, _: ?usize) void {
             const run_node: *RunQueue.Node = @fieldParentPtr("data", runnable);
             const closure: *@This() = @alignCast(@fieldParentPtr("run_node", run_node));
             @call(.auto, func, closure.arguments);

